frontend/src/components/admin/UserCard.tsx

```diff
@@ -27,12 +42,42 @@ const DefaultUserIcon = ({
 };
 
 const UserCard = ({
+  id,
   firstName,
   lastName,
   role,
   email,
   town,
 }: UserCardProps): React.ReactElement => {
+  const userCardOptions = ["User", "Admin", "Staff"];
+  const [selectValue, setSelectValue] = useState(userCardOptions[0]);
+  const [currRole, setCurrRole] = useState(role);
```

Comment:
This solution almost works, but unfortunately, if you switch users (email search) and switch back, the role does not seem to update properly. It might be simpler instead to just make use of `refetchQueries`.